### ISSUE TITLE
[Feature] More configurable request timeouts

### ISSUE THREAD

**Issue by awinder** [usebruno/bruno#1249]
Previously there was some work done in #557 to add configuration for request timeouts, however, the code addressed only some of what was discussed in that issue. I think it would be nice to:

1. Add support to the .bru / collection.bru files to specify a timeout
2. Handle the timeout settings in the bruno CLI
3. Allow a `--timeout` override option on the CLI (useful for CI environments where you might have different timing parameters than in local dev, for instance)
4. Add UI support for setting these timeout values

I have code done for 1-3, wondering about the UI though. Maybe "timeout" could go under the Presets tab in collection settings? I'm not sure where a good place would be for this config in the request settings though. Vars might be the closest tab for something like this but even then it's not great, and adding another tab just for the timeout seems a little much. Any thoughts on what would make sense here? Is this feature useful? I could also punt on making the UI changes but I assume it's best to ship support for stuff like this with UI at the same time as it's enabled by the underlying system / config files. 

#557 also mentioned being able to configure this on the environment level, but I'm not sure how that would work because the environment configuration is all substitution variables, there's not something like a direct config override. Maybe that could work if the timeout settings in collection + request accepted variable substitution? Wondering if that's worth pursuing or not as well.

**Comment by awinder** [comment]
I've taken a crack at this in #1308 with the UI from point #4 implemented by renaming the Docs tab to "Meta" and pushing docs + this new option under that tab. I did punt on including variable support / something tied into environments for now but if that's a showstopper we could discuss as well. @helloanoop -- mind checking this out / discussing?

**Comment by usrbad** [comment]
Hi guys. So there's no way to set timeout now in CLI right?

**Comment by awinder** [comment]
Correct, it’s in the linked pull request but never made it to master. I’d be happy to update / resolve merge conflicts if project maintainers have interest in this. I tried keeping it up to date for a few months but couldn’t get traction to have someone review the feature.

**Comment by frille7070** [comment]
I would VERY much appreciate the possibility to configure timeouts in CLI, like --timeout. Please, if possible.

**Comment by ko-maren** [comment]
Is there any update regarding configuring the timeout via cli?

**Comment by NoPatient** [comment]
I'm really looking forward to this feature!

**Comment by helloanoop** [comment]
Bruno now supports request level timeouts in the UI.
Support in CLI is pending. We've allocated time for this in December. 

Happy to review a PR if anyone in the community wants to take a stab at it.

<img width="549" height="567" alt="Image" src="https://github.com/user-attachments/assets/b0d8d810-ebe7-49ad-8d8a-c687e40abc33" />

**Comment by awinder** [comment]
@helloanoop this is cool / much better than what I tried to do originally for per-request settings. I put up a pull request for just CLI support for a `--max-timeout` flag here:

https://github.com/usebruno/bruno/pull/5847

thought with `--max-timeout` flag was that it seemed more logical to involve any per-request timeout configs in the logic for how to apply to the CLI. So with `--max-timeout` I could make sure that no request went on way too long, while also enforcing that sensitive endpoints returned quickly or else threw errors. totally willing to change this logic if something else makes sense 😄

**Comment by finbargp** [comment]
Just checking if there's any update on CLI support for timeout?

**Comment by NoPatient** [comment]
Hi dear, I look forward to this feature quite a long time. Could I help any in this issue?

**Comment by hima-bash** [comment]
Is there a way now to increase the timeout in the *.bru files ?
